src/Tasks/Hash.cs

```diff
@@ -37,28 +38,42 @@ public override bool Execute()
         {
             if (ItemsToHash != null && ItemsToHash.Length > 0)
             {
-                StringBuilder hashInput = new StringBuilder();
-
-                foreach (var item in ItemsToHash)
-                {
-                    hashInput.Append(item.ItemSpec);
-                    hashInput.Append(ItemSeparatorCharacter);
-                }
-
                 using (var sha1 = SHA1.Create())
                 {
-                    var hash = sha1.ComputeHash(Encoding.UTF8.GetBytes(hashInput.ToString()));
-                    var hashResult = new StringBuilder(hash.Length*2);
+                    byte[] hash;
 
-                    foreach (byte b in hash)
+                    var totalItemSize = 0;
+
+                    foreach (var item in ItemsToHash)
                     {
-                        hashResult.Append(b.ToString("x2"));
+                        totalItemSize += item.ItemSpec.Length;
                     }
 
-                    HashResult = hashResult.ToString();
+                    using (var stringBuilder = new ReuseableStringBuilder(totalItemSize + ItemsToHash.Length))
+                    {
+                        foreach (var item in ItemsToHash)
+                        {
+                            stringBuilder.Append(item.ItemSpec);
+                            stringBuilder.Append(ItemSeparatorCharacter);
+                        }
+
+                        //todo return the string builder bytes when the APIs will get that overload
+                        hash = sha1.ComputeHash(Encoding.UTF8.GetBytes(stringBuilder.ToString()));
+                    }
+
+                    using (var hashResult = new ReuseableStringBuilder(hash.Length * 2))
```

Comment:
Think it's worth including this in the length computation above to use a single stringbuilder for this whole operation?